lib/phoenix/debug.ex

```diff
@@ -0,0 +1,89 @@
+defmodule Phoenix.Debug do
+  @moduledoc """
+  Functions for runtime introspection and debugging of Phoenix applications.
+
+  TODO
+  """
+
+  @doc """
+  Returns a list of all currently connected socket processes.
+
+  Each process corresponds to one connection that can have multiple channels.
+
+  For example, when using Phoenix LiveView, the browser establishes a socket
+  connection when initially navigating to the page, and each live navigation
+  retains the same socket connection. Nested LiveViews also share the same
+  connection, each being a different channel. See `Phoenix.Debug.channels/1`.
+
+  ## Examples
+
+      iex> Phoenix.Debug.list_sockets()
+      [#PID<0.123.0>]
+
+  """
+  def list_sockets do
```

Comment:
I am wondering if this is a misnomer. We are not listing all sockets, but perhaps the ones that have the channels implementation. Someone could still define their own socket with their own protocol and then not reply to our messages.

Given this, I would perhaps focus on providing `list_channels` instead, and get that direct result first. We can revisit and add `list_sockets` later.

Reply:
Do you have an example where we'd expect this to happen? I don't see us expecting someone to define a socket without `use`ing `Phoenix.Socket` anywhere.

Reply:
See the docs for `Phoenix.Socket.Transport` :)

Reply:
Aah I always forget about `Phoenix.Socket.Transport`... Maybe because the naming there seems a bit off to me? For me, the "transport" is WebSocket / Longpoll, but `Phoenix.Socket.Transport` is not really a custom transport, but still on top of WebSocket / Longpoll.

Anyway, I kept the extra function and renamed it to `list_channel_sockets`, because to get the actual channels, I'd either need to list the children of the channel supervisor or send messages to the socket processes and I wanted to keep it simple (filtering Process.list).